### PR TITLE
Improve documentation of date utility packages

### PR DESCRIPTION
**Pull request by markus-da**
Fixes https://github.com/digital-asset/daml-finance/issues/1017

### CODE REVIEW

File: docs/2.8.0/docs/daml-finance/packages/implementations/daml-finance-util.rst [hunk 1]
```diff
@@ -4,8 +4,9 @@
 Daml.Finance.Util
 #################
 
-This package mainly contains utility functions related to dates, lists, maps, and disclosure. They
-are defined in the following modules:
+This package mainly contains utility functions related to dates (see the
```
Comment (brian-weir) [inline docs/2.8.0/docs/daml-finance/packages/implementations/daml-finance-util.rst:7]:
```suggestion
This package primarily contains utility functions related to dates (see the
```

File: docs/2.8.0/docs/daml-finance/packages/interfaces/daml-finance-interface-types-date.rst [hunk 2]
```diff
@@ -4,17 +4,26 @@
 Daml.Finance.Interface.Types.Date
 #################################
 
-This package contains types related to dates. They are defined in the following modules:
+Financial instruments, especially those related to interest rates, often depend on periodic
+schedules. For example, consider a fixed rate bond with a coupon that should be paid every three
+months. Such a regular schedule sometimes results in a payment date falling on a weekend or on
+another holiday. Those dates need to be adjusted to business dates, according to the terms of the
```
Comment (brian-weir) [inline docs/2.8.0/docs/daml-finance/packages/interfaces/daml-finance-interface-types-date.rst:10]:
```suggestion
a holiday. Those dates need to be adjusted to business dates, according to the terms of the
```

**Review by brian-weir — APPROVED** [review]
lgtm - some minor comments

**Review by matteolimberto-da — APPROVED** [review]
Thanks Markus!

Do you think that a tutorial on this package would be a valuable addition to the docs? I feel that it would benefit the users who do not have a strong financial background, as these concepts are not rocket science but need to be explained one by one

**Comment by markus-da** [comment]
> Do you think that a tutorial on this package would be a valuable addition to the docs? I feel that it would benefit the users who do not have a strong financial background, as these concepts are not rocket science but need to be explained one by one

Possibly, I think it depends on who the target user would be for such a tutorial, for example

1. A user of Daml Finance instruments is probably mostly interested in what these concepts mean on a higher level and what they need to specify for a given trade. There is an overview of this functionality in the [bond instrument documentation] (https://docs.daml.com/daml-finance/instruments/bond.html#:~:text=The%20day%20count,initial%20stub%20period.)
2. A developer of Daml Finance instruments probably needs a more in-depth understanding, but such a quant developer cam maybe also be expected to know the concepts beforehand?

In any case, I could link to the relevant part of the bond instrument docs from the Interface.Types.Date doc page for an example on how this is used in practice.

What do you think?

**Comment by matteolimberto-da** [comment]
> > Do you think that a tutorial on this package would be a valuable addition to the docs? I feel that it would benefit the users who do not have a strong financial background, as these concepts are not rocket science but need to be explained one by one
> 
> Possibly, I think it depends on who the target user would be for such a tutorial, for example
> 
> 1. A user of Daml Finance instruments is probably mostly interested in what these concepts mean on a higher level and what they need to specify for a given trade. There is an overview of this functionality in the [bond instrument documentation] (https://docs.daml.com/daml-finance/instruments/bond.html#:~:text=The%20day%20count,initial%20stub%20period.)
> 2. A developer of Daml Finance instruments probably needs a more in-depth understanding, but such a quant developer can maybe also be expected to know the concepts beforehand?
> 
> In any case, I could link to the relevant part of the bond instrument docs from the `Interface.Types.Date` doc page for an example on how this is used in practice.
> 
> What do you think?

Linking to some parts of the bond documentation sounds good.

On your point number 2, I feel that most of our users have a developer background rather than a quant dev background (such as ours). That's why I'm thinking that a tutorial / primer on dates, schedules, calendars, ... would be quite empowering for them. @frankPreiwussDA any thought?

**Comment by markus-da** [comment]
> > > Do you think that a tutorial on this package would be a valuable addition to the docs? I feel that it would benefit the users who do not have a strong financial background, as these concepts are not rocket science but need to be explained one by one
> > 
> > 
> > Possibly, I think it depends on who the target user would be for such a tutorial, for example
> > 
> > 1. A user of Daml Finance instruments is probably mostly interested in what these concepts mean on a higher level and what they need to specify for a given trade. There is an overview of this functionality in the [bond instrument documentation] (https://docs.daml.com/daml-finance/instruments/bond.html#:~:text=The%20day%20count,initial%20stub%20period.)
> > 2. A developer of Daml Finance instruments probably needs a more in-depth understanding, but such a quant developer can maybe also be expected to know the concepts beforehand?
> > 
> > In any case, I could link to the relevant part of the bond instrument docs from the `Interface.Types.Date` doc page for an example on how this is used in practice.
> > What do you think?
> 
> Linking to some parts of the bond documentation sounds good.
> 
> On your point number 2, I feel that most of our users have a developer background rather than a quant dev background (such as ours). That's why I'm thinking that a tutorial / primer on dates, schedules, calendars, ... would be quite empowering for them. @frankPreiwussDA any thought?

I opened an issue for it and added it to the UX roadmap, so we can discuss it in the next UX session.
https://github.com/digital-asset/daml-finance/issues/1044

**Comment by matteolimberto-da** [comment]
Thanks!

**Pull request merged**